skel/handlers/middlewares.go

```diff
@@ -12,3 +16,73 @@ func KaigaraConfigMiddleware(config *kaigara.KaigaraConfig) gin.HandlerFunc {
 		c.Next()
 	}
 }
+
+// AuthMiddleware middleware to verify bearer token
+func AuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Get bearer token from header
+		authHeader := strings.Split(c.GetHeader("Authorization"), "Bearer ")
+		if len(authHeader) != 2 {
+			c.Abort()
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Authorization header not found"})
+			return
+		}
+
+		// Get PublicKey from header
+		publicKey := c.GetHeader("PublicKey")
+		if publicKey == "" {
+			c.Abort()
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"error": "PublicKey is missing in header"})
+			return
+		}
+
+		// Load public key
+		keyStore := jwt.KeyStore{}
+		keyStore.LoadPublicKeyFromString(publicKey)
+
+		// Parse token
+		jwtToken := authHeader[1]
+		auth, err := jwt.ParseAndValidate(jwtToken, keyStore.PublicKey)
+		if err != nil {
+			c.Abort()
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Unauthorized"})
+			return
+		}
+
+		c.Set("auth", &auth)
+
+		c.Next()
+	}
+}
+
+// AdminMiddleware middleware to verity admin role
+func AdminMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		auth, err := GetAuth(c)
+		if err != nil {
+			c.Abort()
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Unauthorized"})
+			return
+		}
+
+		if !isAdminRole(auth.Role) {
+			c.Abort()
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Unauthorized"})
+			return
+		}
+
+		c.Next()
+	}
+}
+
+func isAdminRole(role string) bool {
+	var roles = []string{"superadmin", "admin", "accountant", "compliance", "support", "technical"}
```

Comment:
we need to think about this one, maybe this list is good enough as a first filter for the /api/v2/admin/*
then for the registration endpoint we need to strictly limit to superadmin only.

Reply:
I will keep this function and add more check for only "superadmin" role in CreatePlatform handler.

Is it ok?